File: AILang/benchmarks/c/format_print.c
#include <stdint.h>
#include <stdio.h>

static int64_t format_print_bench(int64_t iterations) {
    int64_t sink = 0;
    for (int64_t i = 0; i < iterations; i++) {
        printf("%lld\n", (long long)i);
        sink += i;
    }
    return sink;
}

int main(void) {
    const int64_t iterations = 100;
    int64_t result = format_print_bench(iterations);
    printf("%lld\n", (long long)result);
    return 0;
}

File: AILang/benchmarks/rust/format_print.rs
fn format_print_bench(iterations: i64) -> i64 {
    let mut sink: i64 = 0;
    let mut i: i64 = 0;
    while i < iterations {
        println!("{}", i);
        sink += i;
        i += 1;
    }
    sink
}

fn main() {
    let iterations: i64 = 100;
    let result = format_print_bench(iterations);
    println!("{}", result);
}
